runners/direct-java/src/main/java/org/apache/beam/runners/direct/StepTransformResult.java

```diff
@@ -127,5 +148,10 @@ public Builder addOutput(Collection<UncommittedBundle<?>> outputBundles) {
       bundlesBuilder.addAll(outputBundles);
       return this;
     }
+
+    public Builder withAdditionalOutput(boolean producedAdditionalOutput) {
```

Comment:
This has been changed to a set of types of outputs that were produced. Committing a result determines if the evaluation produced any bundles.